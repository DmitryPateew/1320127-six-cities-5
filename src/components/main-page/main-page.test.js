import React from "react";
import {MainPage} from "./main-page";
import {expect, it} from "@jest/globals";
import Enzyme, {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

const offersTest = [{
  id: 1,
  city: `Hamburg`,
  coords: [52.3909553943508, 4.85309666406198],
  favorite: true,
  photos: [`img/apartment-02.jpg`, `img/apartment-01.jpg`],
  title: `Beautiful & luxurious studio at great location`,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
  premium: false,
  typeOfHousing: `room`,
  rating: 1,
  countBedrooms: 3,
  maxAdults: 3,
  pricePerNight: 150,
  facilities: [`Wi-Fi`, `Heating`, `Kitchen`],
}, {
  id: 2,
  city: `Hamburg`,
  coords: [52.3909553943508, 4.85309666406198],
  favorite: true,
  photos: [`img/apartment-02.jpg`, `img/apartment-01.jpg`],
  title: `Beautiful & luxurious studio at great location`,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
  premium: false,
  typeOfHousing: `room`,
  rating: 1,
  countBedrooms: 3,
  maxAdults: 3,
  pricePerNight: 150,
  facilities: [`Wi-Fi`, `Heating`, `Kitchen`]
}];

it(`Should MainPage render correctly`, () => {
  const mainPage = shallow(<MainPage
    changeOverId={() => {
    }}
    changeFilter={() => {
    }}
    changeCity={() => {
    }}
    offers={offersTest}
    activeFilter={`To Low`}
    activeCity={`NewYork`}
    toggle={false}
    changeToggleFilter={() => {
    }}
  />);

  expect(shallowToJson(mainPage)).toMatchSnapshot();
});
