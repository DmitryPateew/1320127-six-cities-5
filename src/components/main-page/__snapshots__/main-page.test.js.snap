// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MainPage render correctly 1`] = `
<div
  className="page page--gray page--main"
>
  <Connect(Header) />
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <CityList
      activeCity="NewYork"
      changeCity={[Function]}
    />
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
             
            0
             places to stay in 
            NewYork
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
            onClick={[Function]}
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              className="places__sorting-type"
              tabIndex="0"
            >
              To Low
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              />
            </span>
            <SortList
              activeFilter="To Low"
              changeFilter={[Function]}
              toggle={false}
            />
          </form>
          <Connect(OfferList)
            activeCity="NewYork"
            activeFilter="To Low"
            changeOverId={[Function]}
          />
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <Connect(Map) />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
